server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -84,10 +84,6 @@
      * Returns the mapped field type for the given field name.
      */
     MappedFieldType get(String field) {
-        if (field.equals(TypeFieldType.NAME)) {
```

Comment:
I think this means we can remove the 'type' field and constructor parameter as well?

Reply:
Oops, yep I'll remove it.